### ISSUE TITLE
Does not work with Mac OS Big Sur (11.0.1)

### ISSUE THREAD

**Issue by mitnal** [erickguan/ffi-icu#44]
Hi, I think the Problem is that Big Sur has a built-in dynamic linker (which I do not find any information about except some outdated links to some beta release notes, see below)

Error Message:
```
/ffi-icu-0.2.0/lib/ffi-icu/lib.rb:54:in `load_icu': Could not find ICU on :osx. Patches welcome, or you can add the containing directory yourself: ICU::Lib.search_paths << '/path/to/lib' (LoadError)
```
I played around a little bit with loading of the library and I think it loads without problem with just the filename (without path). But the file does not exist anywhere on disk.

```ruby
module SomeLib
  extend FFI::Library
end

lib = SomeLib.ffi_lib('libicucore.dylib')
# => [#<FFI::DynamicLibrary:0x00007fedd2995940 @name="libicucore.dylib">]

lib.find_function 'u_errorName'
# => #<FFI::Library::Symbol name=u_errorName address=0x00007fff225f43f8>
```

I have no clue how to best check for this 😞. I think we would need to skip the file system check in [here](https://github.com/erickguan/ffi-icu/blob/master/lib/ffi-icu/lib.rb#L29-L33) somehow but only for Big Sur.

See this tweet: https://twitter.com/mycoliza/status/1275305517066227712
And this (at least for me not working) link: https://developer.apple.com/documentation/macos-release-notes/macos-big-sur-11-beta-release-notes
Other Projects with the same problem (and reference of not working link): https://gitlab.kitware.com/cmake/cmake/-/issues/20863

**Comment by Buju77** [comment]
hi,
we are having the same issue. @mitnal thanks for all the links. I did some googling for how to check for os version and found this: https://www.ruby-forum.com/t/detect-mac-os-x-version/125646/2

so maybe we can do something like the following:

```ruby
def self.load_icu
  # First find the library
  lib_names = case ICU.platform
              when :osx
                # check for macOS Big Sur (11.0+)
                os_version = `sw_vers -productVersion` # use terminal command to fetch OS version
                if Gem::Version.new(os_version) >= Gem::Version.new("11.0")
                  # load lib the new way for Big Sur and onwards
                  ...
                else
                  # macOS < 11.0
                  [find_lib("libicucore.#{FFI::Platform::LIBSUFFIX}")]
                end
   ...
end
```

but I have no idea how the new way of loading a lib looks like in Big Sur :(

**Comment by maksym-lebedynets** [comment]
hello everybody. We have also faced this issue.
I see an opened MR there which is going to fix the issue. @erickguan are you going to merge it?

**Comment by CT014330** [comment]
The fix in #45 worked for me locally, would be great to get it merged!

**Comment by damiann** [comment]
Fixed in [v0.3.0](https://github.com/erickguan/ffi-icu/tree/v0.3.0)
